sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BigQueryWriteIT.java

```diff
@@ -0,0 +1,227 @@
+
+    String createTableStatement =
+        "CREATE TABLE ORDERS( \n"
+        + "   id BIGINT, \n"
+        + "   name VARCHAR ) \n"
+        + "TYPE 'bigquery' \n"
+        + "LOCATION '" + bigQueryLocation + "'";
+    executeCreateTable(sqlEnv, createTableStatement);
+
+    executeInsert(pipeline, sqlEnv, "INSERT INTO ORDERS VALUES (1, 'foo')");
+
+    pipeline.run().waitUntilFinish(Duration.standardMinutes(2));
+
+    PAssert
+        .that(allRowsFromBQ(resultsPipeline, bigQueryLocation))
+        .containsInAnyOrder(row(RESULT_SCHEMA, "1", "foo"));
+    resultsPipeline.run().waitUntilFinish(Duration.standardMinutes(2));
+  }
+
+  @Test
+  public void testInsertSelect() throws Exception {
+    String bigQueryLocation = createBQTableName("testInsertSelect");
+
+    BeamSqlEnv sqlEnv = newSqlEnv(
+        inMemoryTableProvider(
+            pipeline,
+            "ORDERS_IN_MEMORY",
+            row(SOURCE_SCHEMA, 1L, "foo"),
+            row(SOURCE_SCHEMA, 2L, "bar"),
+            row(SOURCE_SCHEMA, 3L, "baz")),
+        new BigQueryTableProvider()
+    );
+
+    String createTableStatement =
+        "CREATE TABLE ORDERS_BQ( \n"
+        + "   id BIGINT, \n"
+        + "   name VARCHAR ) \n"
+        + "TYPE 'bigquery' \n"
+        + "LOCATION '" + bigQueryLocation + "'";
+    executeCreateTable(sqlEnv, createTableStatement);
+
+    String insertStatement =
+        "INSERT INTO ORDERS_BQ \n"
+        + "SELECT id, name FROM ORDERS_IN_MEMORY";
+    executeInsert(pipeline, sqlEnv, insertStatement);
+
+    pipeline.run().waitUntilFinish(Duration.standardMinutes(2));
+
+    PAssert
+        .that(allRowsFromBQ(resultsPipeline, bigQueryLocation))
+        .containsInAnyOrder(
+            row(RESULT_SCHEMA, "1", "foo"),
+            row(RESULT_SCHEMA, "2", "bar"),
+            row(RESULT_SCHEMA, "3", "baz"));
+    resultsPipeline.run().waitUntilFinish(Duration.standardMinutes(2));
+  }
+
+  private TableProvider inMemoryTableProvider(
+      Pipeline pipeline,
+      String tableName,
+      Row ... rows) {
+
+    return
+        new BeamSqlTableProvider(
+            "PCOLLECTION",
+            ImmutableMap.of(
+                tableName,
+                new BeamPCollectionTable(createPCollection(pipeline, rows))));
+  }
+
+  private PCollection<Row> createPCollection(Pipeline pipeline, Row ... rows) {
+    return pipeline.apply(Create.of(Arrays.asList(rows)).withCoder(SOURCE_SCHEMA.getRowCoder()));
+  }
+
+  private String createBQTableName(String testName) throws Exception {
+    TestBigQueryOptions pipelineOptions =
+        TestPipeline.testingPipelineOptions().as(TestBigQueryOptions.class);
+    String project = pipelineOptions.getProject();
+    String dataSet = pipelineOptions.getIntegrationTestDataset();
+    String table = createRandomizedName(testName);
+    return String.format("%s:%s.%s", project, dataSet, table);
+  }
+
+  private PCollection<Row> allRowsFromBQ(Pipeline pipeline, String bigQueryLocation) {
```

Comment:
(nit) Would make a composite PTransform from PBegin to PCollection<Row>. Doesn't matter that much for test.